src/Data/NaturalTransformation.purs

```diff
@@ -14,6 +14,7 @@ module Data.NaturalTransformation where
 -- | `f` and `g` should be functors, but the `Functor` constraint is not
 -- | enforced here; that the types are of kind `Type -> Type` is enough for our
 -- | purposes.
+type NaturalTransformation :: forall k. (k -> Type) -> (k -> Type) -> Type
```

Comment:
This kind signature disagrees with the doc-comments. I feel like perhaps we should make this `(Type -> Type) -> (Type -> Type) -> Type` unless there is a specific examples where the generalized version would be useful.

Reply:
(Or alternatively update the doc comments)

Reply:
That's a good point, I've updated the doc comments.